multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -84,20 +86,24 @@ def is_categorical_column(
                 ratio = 0.1
         threshold = min(int(len(data) * ratio), threshold)
         data_value_counts = data.value_counts(dropna=False)
-        key_set = set(data_value_counts.keys())
-        if len(data_value_counts) < threshold:
-            valid_value_counts = valid_data.value_counts(dropna=False)
-            total_valid_num = len(valid_data)
-            oov_num = 0
-            for k, v in zip(valid_value_counts.keys(), valid_value_counts.values):
-                if k not in key_set:
-                    oov_num += v
-            if is_label and oov_num != 0:
-                return False
-            if oov_num / total_valid_num > oov_ratio_threshold:
+        try:
+            key_set = set(data_value_counts.keys())
+            if len(data_value_counts) < threshold:
+                valid_value_counts = valid_data.value_counts(dropna=False)
+                total_valid_num = len(valid_data)
+                oov_num = 0
+                for k, v in zip(valid_value_counts.keys(), valid_value_counts.values):
+                    if k not in key_set:
+                        oov_num += v
+                if is_label and oov_num != 0:
+                    return False
+                if oov_num / total_valid_num > oov_ratio_threshold:
+                    return False
+                return True
+            else:
                 return False
-            return True
-        return False
+        except:
```

Comment:
Bytearray can't be hashed, so that `set(data_value_counts.keys())` will error out when the column contains bytearray. This function is checking if column contains categorical values, so in this case we just want to return `False`.

Reply:
Instead of adding this try-except, how about using `is_imagebytearray_column ` ahead of `is_categorical_column `?

Reply:
That should work, but I'm curious on the time taken on 500 failed attempts to open images versus other checks if we move `is_imagebytearray_column` near the top of the logic flow. Will it take more time in general, or if this is negligible?

Reply:
Make sense.